[wasm] Use `g_printerr` to emit errors in mini-wasm.c, so they show up wit…

radical opened the pull request:
…h traces in the js console.

With just printf, you would see the error as:

`ThreadPool Callback threw an unhandled exception of type System.TypeInitializationException`

.. but with g_error, you get:

<img width="834" alt="Screen Shot 2021-03-18 at 4 59 57 PM" src="https://user-images.githubusercontent.com/1472/111702992-667a9d80-8813-11eb-8875-c4015fe92f8a.png">

You don't get the message string itself, but that can be found by looking at the source line. Or you can build the runtime with `/p:MonoEnableAssertMessages=true`.

But you do get the stack trace for that error!

radical committed:
[wasm] Use g_error to emit errors in mini-wasm.c, so they show up with traces in js console

radical commented on src/mono/mono/mini/mini-wasm.c:
```diff
@@ -218,7 +218,7 @@ mono_arch_create_vars (MonoCompile *cfg)
 		cfg->vret_addr = mono_compile_create_var (cfg, mono_get_int_type (), OP_ARG);
 		if (G_UNLIKELY (cfg->verbose_level > 1)) {
-			printf ("vret_addr = ");
+			g_error ("vret_addr = ");
```
this one should maybe use `g_log`?

vargaz replied:
printf in JIT code is mapped to the platform logging operation, so this one can stay as it is.

radical committed:
address review feedback

vargaz: printf in JIT code is mapped to the platform logging operation, so this one can stay as it is.

lewing reviewed (commented):
Nice, stack is more useful than a printf.
- fix g_assert to print a stack?
- do something intltool-like and start moving strings to a table we can link in or out at WasmNativeBuild time?

lambdageek commented:
Is the intention here to turn a benign error into a fatal one?  `g_error` aborts the process.

lewing commented:
@lambdageek makes a (very) good point perhaps, use g_print_err like https://github.com/dotnet/runtime/pull/49828

radical committed:
Use g_printerr instead of g_error, as suggested by feedback

radical commented:
> Nice, stack is more useful than a printf.
> 
> * fix g_assert to print a stack?

It gets logged with the stack trace, I confirmed.

> * do something intltool-like and start moving strings to a table we can link in or out at WasmNativeBuild time?

https://github.com/dotnet/runtime/issues/49888
